From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micka=C3=ABl=20Sala=C3=BCn?= <clipos@ssi.gouv.fr>
Date: Mon, 16 Apr 2012 16:40:48 +0200
Subject: [PATCH 09/10] Add 39 new ASN.1 OIDs

The OID values are from OpenSSL (crypto/objects/objects.txt).
---
 src/libstrongswan/asn1/oid.h             | 460 +++++++++++++++++--------------
 src/libstrongswan/asn1/oid.txt           |  38 +++
 src/libstrongswan/utils/identification.c |  40 ++-
 3 files changed, 326 insertions(+), 212 deletions(-)

diff --git a/src/libstrongswan/asn1/oid.h b/src/libstrongswan/asn1/oid.h
index bbd7b93ab721..16df90e91f18 100644
--- a/src/libstrongswan/asn1/oid.h
+++ b/src/libstrongswan/asn1/oid.h
@@ -30,217 +30,255 @@ extern const oid_t oid_names[];
 #define OID_COUNTRY							23
 #define OID_LOCALITY						24
 #define OID_STATE_OR_PROVINCE				25
-#define OID_ORGANIZATION					26
-#define OID_ORGANIZATION_UNIT				27
-#define OID_TITLE							28
-#define OID_DESCRIPTION						29
-#define OID_USER_CERTIFICATE				30
-#define OID_NAME							31
-#define OID_GIVEN_NAME						32
-#define OID_INITIALS						33
-#define OID_UNIQUE_IDENTIFIER				34
-#define OID_DN_QUALIFIER					35
-#define OID_ROLE							36
-#define OID_SUBJECT_KEY_ID					39
-#define OID_KEY_USAGE						40
-#define OID_SUBJECT_ALT_NAME				42
-#define OID_BASIC_CONSTRAINTS				44
-#define OID_CRL_NUMBER						45
-#define OID_CRL_REASON_CODE					46
-#define OID_DELTA_CRL_INDICATOR				49
-#define OID_ISSUING_DIST_POINT				50
-#define OID_NAME_CONSTRAINTS				52
-#define OID_CRL_DISTRIBUTION_POINTS			53
-#define OID_CERTIFICATE_POLICIES			54
-#define OID_ANY_POLICY						55
-#define OID_POLICY_MAPPINGS					56
-#define OID_AUTHORITY_KEY_ID				57
-#define OID_POLICY_CONSTRAINTS				58
-#define OID_EXTENDED_KEY_USAGE				59
-#define OID_FRESHEST_CRL					61
-#define OID_INHIBIT_ANY_POLICY				62
-#define OID_TARGET_INFORMATION				63
-#define OID_NO_REV_AVAIL					64
-#define OID_CAMELLIA128_CBC					75
-#define OID_CAMELLIA192_CBC					76
-#define OID_CAMELLIA256_CBC					77
-#define OID_RSA_ENCRYPTION					90
-#define OID_MD2_WITH_RSA					91
-#define OID_MD5_WITH_RSA					92
-#define OID_SHA1_WITH_RSA					93
-#define OID_RSAES_OAEP						94
-#define OID_SHA256_WITH_RSA					97
-#define OID_SHA384_WITH_RSA					98
-#define OID_SHA512_WITH_RSA					99
-#define OID_SHA224_WITH_RSA					100
-#define OID_PBE_MD5_DES_CBC					102
-#define OID_PBE_SHA1_DES_CBC				103
-#define OID_PBKDF2							104
-#define OID_PBES2							105
-#define OID_PKCS7_DATA						107
-#define OID_PKCS7_SIGNED_DATA				108
-#define OID_PKCS7_ENVELOPED_DATA			109
-#define OID_PKCS7_SIGNED_ENVELOPED_DATA		110
-#define OID_PKCS7_DIGESTED_DATA				111
-#define OID_PKCS7_ENCRYPTED_DATA			112
-#define OID_EMAIL_ADDRESS					114
-#define OID_UNSTRUCTURED_NAME				115
-#define OID_PKCS9_CONTENT_TYPE				116
-#define OID_PKCS9_MESSAGE_DIGEST			117
-#define OID_PKCS9_SIGNING_TIME				118
-#define OID_CHALLENGE_PASSWORD				120
-#define OID_UNSTRUCTURED_ADDRESS			121
-#define OID_EXTENSION_REQUEST				122
-#define OID_X509_CERTIFICATE				125
-#define OID_PBE_SHA1_RC4_128				129
-#define OID_PBE_SHA1_RC4_40					130
-#define OID_PBE_SHA1_3DES_CBC				131
-#define OID_PBE_SHA1_3DES_2KEY_CBC			132
-#define OID_PBE_SHA1_RC2_CBC_128			133
-#define OID_PBE_SHA1_RC2_CBC_40				134
-#define OID_P12_KEY_BAG						137
-#define OID_P12_PKCS8_KEY_BAG				138
-#define OID_P12_CERT_BAG					139
-#define OID_P12_CRL_BAG						140
-#define OID_MD2								144
-#define OID_MD5								145
-#define OID_3DES_EDE_CBC					147
-#define OID_EC_PUBLICKEY					151
-#define OID_C2PNB163V1						154
-#define OID_C2PNB163V2						155
-#define OID_C2PNB163V3						156
-#define OID_C2PNB176W1						157
-#define OID_C2PNB191V1						158
-#define OID_C2PNB191V2						159
-#define OID_C2PNB191V3						160
-#define OID_C2PNB191V4						161
-#define OID_C2PNB191V5						162
-#define OID_C2PNB208W1						163
-#define OID_C2PNB239V1						164
-#define OID_C2PNB239V2						165
-#define OID_C2PNB239V3						166
-#define OID_C2PNB239V4						167
-#define OID_C2PNB239V5						168
-#define OID_C2PNB272W1						169
-#define OID_C2PNB304W1						170
-#define OID_C2PNB359V1						171
-#define OID_C2PNB368W1						172
-#define OID_C2PNB431R1						173
-#define OID_PRIME192V1						175
-#define OID_PRIME192V2						176
-#define OID_PRIME192V3						177
-#define OID_PRIME239V1						178
-#define OID_PRIME239V2						179
-#define OID_PRIME239V3						180
-#define OID_PRIME256V1						181
-#define OID_ECDSA_WITH_SHA1					183
-#define OID_ECDSA_WITH_SHA224				185
-#define OID_ECDSA_WITH_SHA256				186
-#define OID_ECDSA_WITH_SHA384				187
-#define OID_ECDSA_WITH_SHA512				188
-#define OID_MS_SMARTCARD_LOGON				202
-#define OID_USER_PRINCIPAL_NAME				203
-#define OID_STRONGSWAN						209
-#define OID_BLISS_PUBLICKEY					214
-#define OID_BLISS_I							216
-#define OID_BLISS_II						217
-#define OID_BLISS_III						218
-#define OID_BLISS_IV						219
-#define OID_BLISS_WITH_SHA512				221
-#define OID_TCGID							228
-#define OID_BLOWFISH_CBC					232
-#define OID_AUTHORITY_INFO_ACCESS			276
-#define OID_IP_ADDR_BLOCKS					278
-#define OID_POLICY_QUALIFIER_CPS			280
-#define OID_POLICY_QUALIFIER_UNOTICE		281
-#define OID_SERVER_AUTH						283
-#define OID_CLIENT_AUTH						284
-#define OID_OCSP_SIGNING					291
-#define OID_XMPP_ADDR						297
-#define OID_AUTHENTICATION_INFO				301
-#define OID_ACCESS_IDENTITY					302
-#define OID_CHARGING_IDENTITY				303
-#define OID_GROUP							304
-#define OID_OCSP							307
-#define OID_BASIC							308
-#define OID_NONCE							309
-#define OID_CRL								310
-#define OID_RESPONSE						311
-#define OID_NO_CHECK						312
-#define OID_ARCHIVE_CUTOFF					313
-#define OID_SERVICE_LOCATOR					314
-#define OID_CA_ISSUERS						315
-#define OID_IKE_INTERMEDIATE				320
-#define OID_DES_CBC							324
-#define OID_SHA1							325
-#define OID_SHA1_WITH_RSA_OIW				326
-#define OID_ECGDSA_PUBKEY					345
-#define OID_ECGDSA_SIG_WITH_RIPEMD160		348
-#define OID_ECGDSA_SIG_WITH_SHA1			349
-#define OID_ECGDSA_SIG_WITH_SHA224			350
-#define OID_ECGDSA_SIG_WITH_SHA256			351
-#define OID_ECGDSA_SIG_WITH_SHA384			352
-#define OID_ECGDSA_SIG_WITH_SHA512			353
-#define OID_SECT163K1						376
-#define OID_SECT163R1						377
-#define OID_SECT239K1						378
-#define OID_SECT113R1						379
-#define OID_SECT113R2						380
-#define OID_SECT112R1						381
-#define OID_SECT112R2						382
-#define OID_SECT160R1						383
-#define OID_SECT160K1						384
-#define OID_SECT256K1						385
-#define OID_SECT163R2						386
-#define OID_SECT283K1						387
-#define OID_SECT283R1						388
-#define OID_SECT131R1						389
-#define OID_SECT131R2						390
-#define OID_SECT193R1						391
-#define OID_SECT193R2						392
-#define OID_SECT233K1						393
-#define OID_SECT233R1						394
-#define OID_SECT128R1						395
-#define OID_SECT128R2						396
-#define OID_SECT160R2						397
-#define OID_SECT192K1						398
-#define OID_SECT224K1						399
-#define OID_SECT224R1						400
-#define OID_SECT384R1						401
-#define OID_SECT521R1						402
-#define OID_SECT409K1						403
-#define OID_SECT409R1						404
-#define OID_SECT571K1						405
-#define OID_SECT571R1						406
-#define OID_AES128_CBC						415
-#define OID_AES128_GCM						416
-#define OID_AES128_CCM						417
-#define OID_AES192_CBC						418
-#define OID_AES192_GCM						419
-#define OID_AES192_CCM						420
-#define OID_AES256_CBC						421
-#define OID_AES256_GCM						422
-#define OID_AES256_CCM						423
-#define OID_SHA256							425
-#define OID_SHA384							426
-#define OID_SHA512							427
-#define OID_SHA224							428
-#define OID_NS_REVOCATION_URL				442
-#define OID_NS_CA_REVOCATION_URL			443
-#define OID_NS_CA_POLICY_URL				444
-#define OID_NS_COMMENT						445
-#define OID_EMPLOYEE_NUMBER					448
-#define OID_PKI_MESSAGE_TYPE				454
-#define OID_PKI_STATUS						455
-#define OID_PKI_FAIL_INFO					456
-#define OID_PKI_SENDER_NONCE				457
-#define OID_PKI_RECIPIENT_NONCE				458
-#define OID_PKI_TRANS_ID					459
-#define OID_TPM_MANUFACTURER				465
-#define OID_TPM_MODEL						466
-#define OID_TPM_VERSION						467
-#define OID_TPM_ID_LABEL					468
+#define OID_STREET_ADDRESS					26
+#define OID_ORGANIZATION					27
+#define OID_ORGANIZATION_UNIT				28
+#define OID_TITLE							29
+#define OID_DESCRIPTION						30
+#define OID_SEARCH_GUIDE					31
+#define OID_BUSINESS_CATEGORY				32
+#define OID_POSTAL_ADDRESS					33
+#define OID_POSTAL_CODE						34
+#define OID_POST_OFFICE_BOX					35
+#define OID_PHYSICAL_DELIVERY_OFFICENAME	36
+#define OID_TELEPHONE_NUMBER				37
+#define OID_TELEX_NUMBER					38
+#define OID_TELETEX_TERMINAL_IDENTIFIER		39
+#define OID_FACSIMILE_TELEPHONE_NUMBER		40
+#define OID_X121_ADDRESS					41
+#define OID_INTERNATIONAL_ISDN_NUMBER		42
+#define OID_REGISTERED_ADDRESS				43
+#define OID_DESTINATION_INDICATOR			44
+#define OID_PREFERRED_DELIVERY_METHOD		45
+#define OID_PRESENTATION_ADDRESS			46
+#define OID_SUPPORTED_APPLICATION_CONTEXT	47
+#define OID_MEMBER							48
+#define OID_OWNER							49
+#define OID_ROLE_OCCUPANT					50
+#define OID_SEE_ALSO						51
+#define OID_USER_PASSWORD					52
+#define OID_USER_CERTIFICATE				53
+#define OID_CA_CERTIFICATE					54
+#define OID_AUTHORITY_REVOCATION_LIST		55
+#define OID_CERTIFICATE_REVOCATION_LIST		56
+#define OID_CROSS_CERTIFICATE_PAIR			57
+#define OID_NAME							58
+#define OID_GIVEN_NAME						59
+#define OID_INITIALS						60
+#define OID_GENERATION_QUALIFIER			61
+#define OID_UNIQUE_IDENTIFIER				62
+#define OID_DN_QUALIFIER					63
+#define OID_DN_QUALIFIER					64
+#define OID_ENHANCED_SEARCH_GUIDE			65
+#define OID_PROTOCOL_INFORMATION			66
+#define OID_DISTINGUISHED_NAME				67
+#define OID_UNIQUE_MEMBER					68
+#define OID_HOUSE_IDENTIFIER				69
+#define OID_SUPPORTED_ALGORITHMS			70
+#define OID_DELTA_REVOCATION_LIST			71
+#define OID_DMD_NAME						72
+#define OID_PSEUDONYM						73
+#define OID_ROLE							74
+#define OID_SUBJECT_KEY_ID					77
+#define OID_KEY_USAGE						78
+#define OID_SUBJECT_ALT_NAME				80
+#define OID_BASIC_CONSTRAINTS				82
+#define OID_CRL_NUMBER						83
+#define OID_CRL_REASON_CODE					84
+#define OID_DELTA_CRL_INDICATOR				87
+#define OID_ISSUING_DIST_POINT				88
+#define OID_NAME_CONSTRAINTS				90
+#define OID_CRL_DISTRIBUTION_POINTS			91
+#define OID_CERTIFICATE_POLICIES			92
+#define OID_ANY_POLICY						93
+#define OID_POLICY_MAPPINGS					94
+#define OID_AUTHORITY_KEY_ID				95
+#define OID_POLICY_CONSTRAINTS				96
+#define OID_EXTENDED_KEY_USAGE				97
+#define OID_FRESHEST_CRL					99
+#define OID_INHIBIT_ANY_POLICY				100
+#define OID_TARGET_INFORMATION				101
+#define OID_NO_REV_AVAIL					102
+#define OID_CAMELLIA128_CBC					113
+#define OID_CAMELLIA192_CBC					114
+#define OID_CAMELLIA256_CBC					115
+#define OID_RSA_ENCRYPTION					128
+#define OID_MD2_WITH_RSA					129
+#define OID_MD5_WITH_RSA					130
+#define OID_SHA1_WITH_RSA					131
+#define OID_RSAES_OAEP						132
+#define OID_SHA256_WITH_RSA					135
+#define OID_SHA384_WITH_RSA					136
+#define OID_SHA512_WITH_RSA					137
+#define OID_SHA224_WITH_RSA					138
+#define OID_PBE_MD5_DES_CBC					140
+#define OID_PBE_SHA1_DES_CBC				141
+#define OID_PBKDF2							142
+#define OID_PBES2							143
+#define OID_PKCS7_DATA						145
+#define OID_PKCS7_SIGNED_DATA				146
+#define OID_PKCS7_ENVELOPED_DATA			147
+#define OID_PKCS7_SIGNED_ENVELOPED_DATA		148
+#define OID_PKCS7_DIGESTED_DATA				149
+#define OID_PKCS7_ENCRYPTED_DATA			150
+#define OID_EMAIL_ADDRESS					152
+#define OID_UNSTRUCTURED_NAME				153
+#define OID_PKCS9_CONTENT_TYPE				154
+#define OID_PKCS9_MESSAGE_DIGEST			155
+#define OID_PKCS9_SIGNING_TIME				156
+#define OID_CHALLENGE_PASSWORD				158
+#define OID_UNSTRUCTURED_ADDRESS			159
+#define OID_EXTENSION_REQUEST				160
+#define OID_X509_CERTIFICATE				163
+#define OID_PBE_SHA1_RC4_128				167
+#define OID_PBE_SHA1_RC4_40					168
+#define OID_PBE_SHA1_3DES_CBC				169
+#define OID_PBE_SHA1_3DES_2KEY_CBC			170
+#define OID_PBE_SHA1_RC2_CBC_128			171
+#define OID_PBE_SHA1_RC2_CBC_40				172
+#define OID_P12_KEY_BAG						175
+#define OID_P12_PKCS8_KEY_BAG				176
+#define OID_P12_CERT_BAG					177
+#define OID_P12_CRL_BAG						178
+#define OID_MD2								182
+#define OID_MD5								183
+#define OID_3DES_EDE_CBC					185
+#define OID_EC_PUBLICKEY					189
+#define OID_C2PNB163V1						192
+#define OID_C2PNB163V2						193
+#define OID_C2PNB163V3						194
+#define OID_C2PNB176W1						195
+#define OID_C2PNB191V1						196
+#define OID_C2PNB191V2						197
+#define OID_C2PNB191V3						198
+#define OID_C2PNB191V4						199
+#define OID_C2PNB191V5						200
+#define OID_C2PNB208W1						201
+#define OID_C2PNB239V1						202
+#define OID_C2PNB239V2						203
+#define OID_C2PNB239V3						204
+#define OID_C2PNB239V4						205
+#define OID_C2PNB239V5						206
+#define OID_C2PNB272W1						207
+#define OID_C2PNB304W1						208
+#define OID_C2PNB359V1						209
+#define OID_C2PNB368W1						210
+#define OID_C2PNB431R1						211
+#define OID_PRIME192V1						213
+#define OID_PRIME192V2						214
+#define OID_PRIME192V3						215
+#define OID_PRIME239V1						216
+#define OID_PRIME239V2						217
+#define OID_PRIME239V3						218
+#define OID_PRIME256V1						219
+#define OID_ECDSA_WITH_SHA1					221
+#define OID_ECDSA_WITH_SHA224				223
+#define OID_ECDSA_WITH_SHA256				224
+#define OID_ECDSA_WITH_SHA384				225
+#define OID_ECDSA_WITH_SHA512				226
+#define OID_MS_SMARTCARD_LOGON				240
+#define OID_USER_PRINCIPAL_NAME				241
+#define OID_STRONGSWAN						247
+#define OID_BLISS_PUBLICKEY					252
+#define OID_BLISS_I							254
+#define OID_BLISS_II						255
+#define OID_BLISS_III						256
+#define OID_BLISS_IV						257
+#define OID_BLISS_WITH_SHA512				259
+#define OID_TCGID							266
+#define OID_BLOWFISH_CBC					270
+#define OID_AUTHORITY_INFO_ACCESS			314
+#define OID_IP_ADDR_BLOCKS					316
+#define OID_POLICY_QUALIFIER_CPS			318
+#define OID_POLICY_QUALIFIER_UNOTICE		319
+#define OID_SERVER_AUTH						321
+#define OID_CLIENT_AUTH						322
+#define OID_OCSP_SIGNING					329
+#define OID_XMPP_ADDR						335
+#define OID_AUTHENTICATION_INFO				339
+#define OID_ACCESS_IDENTITY					340
+#define OID_CHARGING_IDENTITY				341
+#define OID_GROUP							342
+#define OID_OCSP							345
+#define OID_BASIC							346
+#define OID_NONCE							347
+#define OID_CRL								348
+#define OID_RESPONSE						349
+#define OID_NO_CHECK						350
+#define OID_ARCHIVE_CUTOFF					351
+#define OID_SERVICE_LOCATOR					352
+#define OID_CA_ISSUERS						353
+#define OID_IKE_INTERMEDIATE				358
+#define OID_DES_CBC							362
+#define OID_SHA1							363
+#define OID_SHA1_WITH_RSA_OIW				364
+#define OID_ECGDSA_PUBKEY					383
+#define OID_ECGDSA_SIG_WITH_RIPEMD160		386
+#define OID_ECGDSA_SIG_WITH_SHA1			387
+#define OID_ECGDSA_SIG_WITH_SHA224			388
+#define OID_ECGDSA_SIG_WITH_SHA256			389
+#define OID_ECGDSA_SIG_WITH_SHA384			390
+#define OID_ECGDSA_SIG_WITH_SHA512			391
+#define OID_SECT163K1						414
+#define OID_SECT163R1						415
+#define OID_SECT239K1						416
+#define OID_SECT113R1						417
+#define OID_SECT113R2						418
+#define OID_SECT112R1						419
+#define OID_SECT112R2						420
+#define OID_SECT160R1						421
+#define OID_SECT160K1						422
+#define OID_SECT256K1						423
+#define OID_SECT163R2						424
+#define OID_SECT283K1						425
+#define OID_SECT283R1						426
+#define OID_SECT131R1						427
+#define OID_SECT131R2						428
+#define OID_SECT193R1						429
+#define OID_SECT193R2						430
+#define OID_SECT233K1						431
+#define OID_SECT233R1						432
+#define OID_SECT128R1						433
+#define OID_SECT128R2						434
+#define OID_SECT160R2						435
+#define OID_SECT192K1						436
+#define OID_SECT224K1						437
+#define OID_SECT224R1						438
+#define OID_SECT384R1						439
+#define OID_SECT521R1						440
+#define OID_SECT409K1						441
+#define OID_SECT409R1						442
+#define OID_SECT571K1						443
+#define OID_SECT571R1						444
+#define OID_AES128_CBC						453
+#define OID_AES128_GCM						454
+#define OID_AES128_CCM						455
+#define OID_AES192_CBC						456
+#define OID_AES192_GCM						457
+#define OID_AES192_CCM						458
+#define OID_AES256_CBC						459
+#define OID_AES256_GCM						460
+#define OID_AES256_CCM						461
+#define OID_SHA256							463
+#define OID_SHA384							464
+#define OID_SHA512							465
+#define OID_SHA224							466
+#define OID_NS_REVOCATION_URL				480
+#define OID_NS_CA_REVOCATION_URL			481
+#define OID_NS_CA_POLICY_URL				482
+#define OID_NS_COMMENT						483
+#define OID_EMPLOYEE_NUMBER					486
+#define OID_PKI_MESSAGE_TYPE				492
+#define OID_PKI_STATUS						493
+#define OID_PKI_FAIL_INFO					494
+#define OID_PKI_SENDER_NONCE				495
+#define OID_PKI_RECIPIENT_NONCE				496
+#define OID_PKI_TRANS_ID					497
+#define OID_TPM_MANUFACTURER				503
+#define OID_TPM_MODEL						504
+#define OID_TPM_VERSION						505
+#define OID_TPM_ID_LABEL					506
 
-#define OID_MAX								469
+#define OID_MAX								507
 
 #endif /* OID_H_ */
diff --git a/src/libstrongswan/asn1/oid.txt b/src/libstrongswan/asn1/oid.txt
index 39bc552d4cdf..a44df5dc5127 100644
--- a/src/libstrongswan/asn1/oid.txt
+++ b/src/libstrongswan/asn1/oid.txt
@@ -24,16 +24,54 @@
     0x06                     "C"						OID_COUNTRY
     0x07                     "L"						OID_LOCALITY
     0x08                     "ST"						OID_STATE_OR_PROVINCE
+    0x09                     "street"						OID_STREET_ADDRESS
     0x0A                     "O"						OID_ORGANIZATION
     0x0B                     "OU"						OID_ORGANIZATION_UNIT
     0x0C                     "T"						OID_TITLE
     0x0D                     "D"						OID_DESCRIPTION
+    0x0E                     "searchGuide"					OID_SEARCH_GUIDE
+    0x0F                     "businessCategory"					OID_BUSINESS_CATEGORY
+    0x10                     "postalAddress"					OID_POSTAL_ADDRESS
+    0x11                     "postalCode"					OID_POSTAL_CODE
+    0x12                     "postOfficeBox"					OID_POST_OFFICE_BOX
+    0x13                     "physicalDeliveryOfficeName"			OID_PHYSICAL_DELIVERY_OFFICENAME
+    0x14                     "telephoneNumber"					OID_TELEPHONE_NUMBER
+    0x15                     "telexNumber"					OID_TELEX_NUMBER
+    0x16                     "teletexTerminalIdentifier"			OID_TELETEX_TERMINAL_IDENTIFIER
+    0x17                     "facsimileTelephoneNumber" 			OID_FACSIMILE_TELEPHONE_NUMBER
+    0x18                     "x121Address"					OID_X121_ADDRESS
+    0x19                     "internationaliSDNNumber"				OID_INTERNATIONAL_ISDN_NUMBER
+    0x1a                     "registeredAddress"				OID_REGISTERED_ADDRESS
+    0x1b                     "destinationIndicator"				OID_DESTINATION_INDICATOR
+    0x1c                     "preferredDeliveryMethod"				OID_PREFERRED_DELIVERY_METHOD
+    0x1d                     "presentationAddress"				OID_PRESENTATION_ADDRESS
+    0x1e                     "supportedApplicationContext"			OID_SUPPORTED_APPLICATION_CONTEXT
+    0x1f                     "member"						OID_MEMBER
+    0x20                     "owner"						OID_OWNER
+    0x21                     "roleOccupant"					OID_ROLE_OCCUPANT
+    0x22                     "seeAlso"						OID_SEE_ALSO
+    0x23                     "userPassword"					OID_USER_PASSWORD
     0x24                     "userCertificate"			OID_USER_CERTIFICATE
+    0x25                     "cACertificate"			OID_CA_CERTIFICATE
+    0x26                     "authorityRevocationList"	OID_AUTHORITY_REVOCATION_LIST
+    0x27                     "certificateRevocationList"	OID_CERTIFICATE_REVOCATION_LIST
+    0x28                     "crossCertificatePair"			OID_CROSS_CERTIFICATE_PAIR
     0x29                     "N"						OID_NAME
     0x2A                     "G"						OID_GIVEN_NAME
     0x2B                     "I"						OID_INITIALS
+    0x2C                     "generationQualifier"		OID_GENERATION_QUALIFIER
     0x2D                     "ID"						OID_UNIQUE_IDENTIFIER
     0x2E                     "dnQualifier"				OID_DN_QUALIFIER
+    0x2E                     "dnQualifier"				OID_DN_QUALIFIER
+    0x2F                     "enhancedSearchGuide"		OID_ENHANCED_SEARCH_GUIDE
+    0x30                     "protocolInformation"		OID_PROTOCOL_INFORMATION
+    0x31                     "distinguishedName"		OID_DISTINGUISHED_NAME
+    0x32                     "uniqueMember"				OID_UNIQUE_MEMBER
+    0x33                     "houseIdentifier"			OID_HOUSE_IDENTIFIER
+    0x34                     "supportedAlgorithms"		OID_SUPPORTED_ALGORITHMS
+    0x35                     "deltaRevocationList"		OID_DELTA_REVOCATION_LIST
+    0x36                     "dmdName"					OID_DMD_NAME
+    0x41                     "pseudonym"				OID_PSEUDONYM
     0x48                     "role"						OID_ROLE
   0x1D                       "id-ce"
     0x09                     "subjectDirectoryAttrs"
diff --git a/src/libstrongswan/utils/identification.c b/src/libstrongswan/utils/identification.c
index b69adf399a84..bb40d9dc7f57 100644
--- a/src/libstrongswan/utils/identification.c
+++ b/src/libstrongswan/utils/identification.c
@@ -90,7 +90,45 @@ static const x501rdn_t x501rdns[] = {
 	{"unstructuredName",	OID_UNSTRUCTURED_NAME,		ASN1_IA5STRING},
 	{"UA",					OID_UNSTRUCTURED_ADDRESS,	ASN1_PRINTABLESTRING},
 	{"unstructuredAddress", OID_UNSTRUCTURED_ADDRESS,	ASN1_PRINTABLESTRING},
-	{"TCGID", 				OID_TCGID,					ASN1_PRINTABLESTRING}
+	{"TCGID", 				OID_TCGID,					ASN1_PRINTABLESTRING},
+	{"street",				OID_STREET_ADDRESS,			ASN1_PRINTABLESTRING},
+	{"searchGuide",			OID_SEARCH_GUIDE,			ASN1_PRINTABLESTRING},
+	{"businessCategory",	OID_BUSINESS_CATEGORY,		ASN1_PRINTABLESTRING},
+	{"postalAddress",		OID_POSTAL_ADDRESS,			ASN1_PRINTABLESTRING},
+	{"postalCode",			OID_POSTAL_CODE,			ASN1_PRINTABLESTRING},
+	{"postOfficeBox",		OID_POST_OFFICE_BOX,		ASN1_PRINTABLESTRING},
+	{"physicalDeliveryOfficeName",	OID_PHYSICAL_DELIVERY_OFFICENAME,	ASN1_PRINTABLESTRING},
+	{"telephoneNumber",		OID_TELEPHONE_NUMBER,		ASN1_PRINTABLESTRING},
+	{"telexNumber",			OID_TELEX_NUMBER,			ASN1_PRINTABLESTRING},
+	{"teletexTerminalIdentifier",	OID_TELETEX_TERMINAL_IDENTIFIER,	ASN1_PRINTABLESTRING},
+	{"facsimileTelephoneNumber",	OID_FACSIMILE_TELEPHONE_NUMBER,		ASN1_PRINTABLESTRING},
+	{"x121Address",			OID_X121_ADDRESS,			ASN1_PRINTABLESTRING},
+	{"internationaliSDNNumber",		OID_INTERNATIONAL_ISDN_NUMBER,		ASN1_PRINTABLESTRING},
+	{"registeredAddress",	OID_REGISTERED_ADDRESS,		ASN1_PRINTABLESTRING},
+	{"destinationIndicator",	OID_DESTINATION_INDICATOR,	ASN1_PRINTABLESTRING},
+	{"preferredDeliveryMethod",	OID_PREFERRED_DELIVERY_METHOD,	ASN1_PRINTABLESTRING},
+	{"presentationAddress",		OID_PRESENTATION_ADDRESS,		ASN1_PRINTABLESTRING},
+	{"supportedApplicationContext",	OID_SUPPORTED_APPLICATION_CONTEXT,	ASN1_PRINTABLESTRING},
+	{"member",				OID_MEMBER,					ASN1_PRINTABLESTRING},
+	{"owner",				OID_OWNER,					ASN1_PRINTABLESTRING},
+	{"roleOccupant",		OID_ROLE_OCCUPANT,			ASN1_PRINTABLESTRING},
+	{"seeAlso",				OID_SEE_ALSO,				ASN1_PRINTABLESTRING},
+	{"userPassword",		OID_USER_PASSWORD,			ASN1_PRINTABLESTRING},
+	{"cACertificate",		OID_CA_CERTIFICATE,			ASN1_PRINTABLESTRING},
+	{"authorityRevocationList",					OID_AUTHORITY_REVOCATION_LIST,	ASN1_PRINTABLESTRING},
+	{"certificateRevocationList",					OID_CERTIFICATE_REVOCATION_LIST,	ASN1_PRINTABLESTRING},
+	{"crossCertificatePair",	OID_CROSS_CERTIFICATE_PAIR,	ASN1_PRINTABLESTRING},
+	{"generationQualifier",	OID_GENERATION_QUALIFIER,	ASN1_PRINTABLESTRING},
+	{"dnQualifier",			OID_DN_QUALIFIER,			ASN1_PRINTABLESTRING},
+	{"enhancedSearchGuide",	OID_ENHANCED_SEARCH_GUIDE,	ASN1_PRINTABLESTRING},
+	{"protocolInformation",	OID_PROTOCOL_INFORMATION,	ASN1_PRINTABLESTRING},
+	{"distinguishedName",	OID_DISTINGUISHED_NAME,		ASN1_PRINTABLESTRING},
+	{"uniqueMember",		OID_UNIQUE_MEMBER,			ASN1_PRINTABLESTRING},
+	{"houseIdentifier",		OID_HOUSE_IDENTIFIER,		ASN1_PRINTABLESTRING},
+	{"supportedAlgorithms",	OID_SUPPORTED_ALGORITHMS,	ASN1_PRINTABLESTRING},
+	{"deltaRevocationList",	OID_DELTA_REVOCATION_LIST,	ASN1_PRINTABLESTRING},
+	{"dmdName",				OID_DMD_NAME,				ASN1_PRINTABLESTRING},
+	{"pseudonym",			OID_PSEUDONYM,				ASN1_PRINTABLESTRING}
 };
 
 /**
-- 
2.5.4

